packages/flame_console/lib/src/controller.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flame/game.dart';
+import 'package:flame_console/flame_console.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class ConsoleState {
+  const ConsoleState({
+    this.showHistory = false,
+    this.commandHistoryIndex = 0,
+    this.commandHistory = const [],
+    this.history = const [],
+    this.cmd = '',
+  });
+
+  final bool showHistory;
+  final int commandHistoryIndex;
+  final List<String> commandHistory;
+  final List<String> history;
+  final String cmd;
+
+  ConsoleState copyWith({
+    bool? showHistory,
+    int? commandHistoryIndex,
+    List<String>? commandHistory,
+    List<String>? history,
+    String? cmd,
+  }) {
+    return ConsoleState(
+      showHistory: showHistory ?? this.showHistory,
+      commandHistoryIndex: commandHistoryIndex ?? this.commandHistoryIndex,
+      commandHistory: commandHistory ?? this.commandHistory,
+      history: history ?? this.history,
+      cmd: cmd ?? this.cmd,
+    );
+  }
+}
+
+class ConsoleController<G extends FlameGame> {
+  ConsoleController({
+    required this.repository,
+    required this.game,
+    required this.scrollController,
+    required this.onClose,
+    required this.commands,
+    ConsoleState state = const ConsoleState(),
+  }) : state = ValueNotifier(state);
+
+  final ValueNotifier<ConsoleState> state;
+  final ConsoleRepository repository;
+  final G game;
+  final VoidCallback onClose;
+  final ScrollController scrollController;
+  final Map<String, ConsoleCommand<G>> commands;
```

Comment:
Isn't is enough to have this as a list? The name of the command is in the `ConsoleCommand` class right?
And from the example, it doesn't look like the key is actually used for anything?